Test rating option schema

suhasdombale123 opened the pull request:
## Why?
Added Test cases for Rating Option Schema 
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- If it delivers a story on the Pivotal tracker, please link to it here. -->

## This change addresses the need by:
<!--- List and detail all changes made in this PR. -->
Added test cases in file schema/rating/rating_option_test.exs
added separate describe funtion for both funtions create_changset/2 and update_changeset/2 

[delivers #163248658]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read [CONTRIBUTING.md][contributing].
- [x] My code follows the [style guidelines][contributing] of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [x] I have updated the documentation wherever necessary.
- [x] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

suhasdombale123 committed:
added test cases for Rating Option Schema

suhasdombale123 committed:
added test cases for rating_option schema

arjun289 commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,79 @@
+      assert validity
+    end
+
+    test "fails? with missing code" do
```
All the `required` related params can be tested in just one test. No need to test failure for every single `key` in schema. 

Just pass empty params object to `create_changeset(%RatingOption{}, %{})` and you will get all the required related errors.

arjun289 commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,79 @@
+    end
+
+    test "fails? without position" do
+      param = Map.delete(@valid_attrs, :position)
```
Not tested in the right manner, first of all, insert data and then try to update it and check, all the changesets are receiving empty structs.

arjun289 commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,79 @@
+  end
+
+  describe "update_changeset/2 " do
+    test "succeeds" do
```
First, insert a record and then try to use `update_changeset` on it.

suhasdombale123 committed:
Made changes in update_changset/2 with use of Exmachina libraries
changed the unneccesory duplicate cases for fais case

SagarKarwande commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,48 @@
+    test "fails? with all empty" do
+      c = %{valid?: validity} = RatingOption.create_changeset(%RatingOption{}, %{})
+      refute validity
+      require IEx
```
Remove `require IEx`

SagarKarwande commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,48 @@
+    test "fails? with invalid parameters" do
+      rating_opt = insert(:rating_option)
+      c = RatingOption.update_changeset(rating_opt, %{value: 2})
+      assert c.changes == %{}
```
If we are testing failure, then we should assert the errors

SagarKarwande commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,48 @@
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+  import Snitch.Factory
+  alias Snitch.Core.Tools.MultiTenancy.Repo
```
No alias required as `Snitch.DataCase` already has aliased

suhasdombale123 committed:
made changes :
1. removed the redundant alias of multitenancy repo
2. restructered the code properly
3. removed the redundant test case on update_changeset failure

suhasdombale123 committed:
Made changes in rating factory :
1. added factory method for rating_option_vote schema.
2. added factory method for review schema.

suhasdombale123 committed:
Added test cases for review schema which will test both the create_changeset of review schema and rating_option_vote schema.
1. added test for create_changeset/2 and update_changeset/2

ashish173 committed:
Merge branch 'develop' into Test-rating-option-schema

Jeshan7 committed:
[#163248658]
This change
  - Added rating_option tests(schema)
  - Refactored the rating.ex (factory)

ashish173 committed:
Merge branch 'develop' into Test-rating-option-schema

arjun289 commented on apps/snitch_core/test/data/schema/rating/rating_option_test.exs:
```diff
@@ -0,0 +1,47 @@
+  @tag rating_option_count: 1
+  describe "create_changeset/2" do
+    test "returns a valid changeset", %{rating_options: [rating_option]} do
+      rating_option = Map.from_struct(rating_option)
```
This is not the right approach. Inserting the `rating_options` and then using the same. Please do it with params.

arjun289 reviewed (commented):
Add, `foreign_key` constraint for `rating` in `rating_option_schema` and test it.

Jeshan7 committed:
[#163248658]
 Prior to this change,
 - foreign_key_constraint for rating_id was not present.
 - Directly rating_options from factory was passed as context in create/2

 This change
 - foreign_key_constraint for rating_id is now present.
 - Wrote test case for foreign_key_constraint,
 - Now params are passes through context in create/2.

Jeshan7 committed:
Merge branch 'Test-rating-option-schema' of https://github.com/aviacommerce/avia into Test-rating-option-schema

Jeshan7 committed:
Merge branch 'develop' into Test-rating-option-schema

ashish173 committed:
Merge branch 'develop' into Test-rating-option-schema
